types/directive.go

```diff
@@ -24,6 +28,11 @@ type DirectiveDefinition struct {
 
 type DirectiveList []*Directive
 
+type DirectiveVisitor interface {
```

Comment:
I don't think this interface actually belongs in the `types` package. This package, according to its docs, is here to map the specification into Go code 1:1. This is an implementation detail and not part of the spec.

We could move this to another public package called `directives` without losing any functionality. This would be especially valuable if we add more code to support directives in the future.

Reply:
I agree